docs/jwt-auth.md

```diff
@@ -0,0 +1,153 @@
+  const decodedTokenData = decodeJwtToken(token)
+  
+  ...
+}
+```
+
+``` python
+def initialize(block_event):
+    token = fetch_jwt_token({}, datetime.now())
+    decoded_token_data = decode_jwt_token(token)
+    
+    ...
+```
+
+In this case `token` would be a JWT and `decodedTokenData` will look something like this:
+
+```
+{
+  "bot-id": "0x13k387b37769ce24236c403e76fc30f01fa774176e1416c861yfe6c07dfef71f",
+  "exp": 1660119443,
+  "iat": 1660119413,
+  "jti": "qkd5cfad-1884-11ed-a5c9-02420a639308",
+  "nbf": 1660119383,
+  "sub": "0x556f8BE42f76c01F960f32CB1936D2e0e0Eb3F4D"
+}
+```
+
+Every field above except for `bot-id` is defined in the [JWT standard](https://auth0.com/docs/secure/tokens/json-web-tokens/json-web-token-claims#registered-claims). Additional claims can also be encoded in the JWT token using the first input `claims`, you also specify when the token should expire as the second input:
+
+``` typescript
+const initialize: Initialize = async () => {
+  const token = await fetchJwtToken({key: "value"}, Date.now())
+  const decodedTokenData = decodeJwtToken(token)
+
+  ...
+}
+```
+
+``` python
+def initialize():
+    token = fetch_jwt_token({'key': 'value'})
+    decoded_token_data = decode_jwt_token(token)
+
+    ...
+```
+
+which will result in a `decodedTokenData` like below:
+
+```
+{
+  "bot-id": "0x13k387b37769ce24236c403e76fc30f01fa774176e1416c861yfe6c07dfef71f",
+  "exp": 1660119443,
+  "iat": 1660119413,
+  "jti": "qkd5cfad-1884-11ed-a5c9-02420a639308",
+  "nbf": 1660119383,
+  "sub": "0x556f8BE42f76c01F960f32CB1936D2e0e0Eb3F4D",
+  "key": "value"
+}
+```
+
+A useful tool for manually decoding tokens is [this web app](https://jwt.io/) provided by auth0.
+
+## Detection Bot Authentication Example
+
+Imagine you have a backend service that exposes an API endpoint that you would like your detection bot be authorized to use. Here is an example express api endpoint that uses a JWT for authentication at the host name of `external-api`. This server has a GET endpoint the returns data from an applications database (this example does not add any additional claims):
+
+``` typescript
+import express, { Request, Response } from "express";
+import { decodeJwtToken } from "forta-agent";
+
+const PATH = `/example-endpoint`;
+const PORT = 3000;
+
+const app = express();
+const requestRouter = express.Router();
+
+interface JwtPayload {
+  botId: string,
+  exp: number,
+  iat: number,
+  jti: string,
+  nbf: number,
+  sub: string
+}
+
+// Define /example-endpoint
+requestRouter.get(PATH, async (request: Request, response: Response) => {
+
+    // Assuming the JWT token is passed in the header "x-access-token". You can choose a different method to pass the JWT
+    const token = req.headers["x-access-token"];
+
+    const data: JwtPayload = decodeJwtToken(token);
+
+    // If you add additional claims such as api keys or secrets your verification logic can use those as well
+    if(data && data.botId === "YOUR_BOT_ID") {
```

Comment:
in addition to these checks, the **signature** of the jwt token would need to be verified to be from a valid scanner address that the bot is currently assigned to. once you extract the scanner address from the jwt signature, you can use the [Forta subgraph endpoint](https://api.thegraph.com/subgraphs/name/forta-network/forta-network) to determine whether the bot is currently assigned to the scanner with the following example query:

```
query {
  links(where: {
    bot_: {
      id: "0x77281ae942ee1fe141d0652e9dad7d001761552f906fb1684b2812603de31049"
    },
    scanner_: {
      id: "0x1058050aaa816eeeb7e1f396f76cbc0e306db6db"
    }
  }) {
    scanner {
      id
      chainId
    }
  }
}
```

Reply:
also, lets validate the expiry time of the jwt

Reply:
Call the `areTheyLinked` method directly on the smart contract for this check